shell/platform/embedder/embedder_external_view.cc

```diff
@@ -59,7 +58,9 @@ bool EmbedderExternalView::HasPlatformView() const {
 }
 
 bool EmbedderExternalView::HasEngineRenderedContents() const {
-  return canvas_spy_->DidDrawIntoCanvas();
+  TryEndRecording();
+  slice_->dispatch(*dl_op_spy_);
+  return dl_op_spy_->DidDraw() && !slice_->is_empty();
```

Comment:
If this method is called more than once it will repeat the op_spy operation. Also, it will re-use the op_spy object which means the second answer might not be correct. I would change this so that it only does the spy operation once and remembers the answer.

Also, you don't need to permanently store an op_spy object, just create one on the fly when you need to compute the answer and remove the dl_op_spy field from the object.